From 37870264c163bc05b2d04f0c07021f6381b24846 Mon Sep 17 00:00:00 2001
From: Yu Jiahua <yujiahua1@huawei.com>
Date: Fri, 15 Oct 2021 16:00:19 +0800
Subject: [PATCH] sched: Fix branch prediction error in static_key
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 37870264c163bc05b2d04f0c07021f6381b24846


hulk inclusion
category: feature
bugzilla: 175551 https://gitee.com/openeuler/kernel/issues/I4DDEL

--------

sched_blocked_averages is trun on in most cases, it should be static_branch_likely
rather than static_branch_unlikely.

Signed-off-by: Yu Jiahua <yujiahua1@huawei.com>
Reviewed-by: Chen Hui <judy.chenhui@huawei.com>
Signed-off-by: Chen Jun <chenjun102@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 kernel/sched/fair.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index 96793d50c621..bd42f8e9f5ad 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -8327,7 +8327,7 @@ static void update_blocked_averages(int cpu)
 	update_rq_clock(rq);
 
 #ifdef CONFIG_SCHED_OPTIMIZE_LOAD_TRACKING
-	if (!static_branch_unlikely(&sched_blocked_averages)) {
+	if (!static_branch_likely(&sched_blocked_averages)) {
 		rq_unlock_irqrestore(rq, &rf);
 		return;
 	}
-- 
2.26.2

